src/ansi-c/ansi_c_entry_point.cpp

```diff
@@ -173,10 +174,25 @@ bool ansi_c_entry_point(
     return false; // give up
   }
 
-  static_lifetime_init(symbol_table, symbol.location);
+  if(enable_static_lifetime_initialisation)
```

Comment:
Complete removal of `__cprover_initialise` instead of nondet initialisation of some of the variables in there. This avoids ever adding pointer typed assignments such as `__CPROVER_memory_leak = NULL;` and it avoids array expressions such as `__CPROVER_threads_exited = ARRAY_OF(FALSE);`.

The idea is that this will facilitate implementation and integration testing of an initial subset of functionality with integer support only from source through to solver, without support for pointers and structs. Support for the missing functionality can then be added afterwards in piece-wise fashion, until we can put `__cprover_initialise` back in place. This will allow us to thoroughly test and debug the areas which are complete rather than attempting to implement everything at once and then hunting the bugs across a broad swathe of functionality.

Reply:
Why not simply ignore certain types in the conversion?  That would be less intrusive and give you better diagnostics.

Reply:
Because we would like to error rather than ignore on encountering unimplemented functionality in the new `exprt` to smt2 conversion in the incremental support. That way we can be sure that all **user** code up to the current point of execution is supported, without having to read back through the output checking for warnings. `__cprover_initialise` conflicts with this approach, because it introduces potentially unimplemented expressions at the beginning of the analysis before we even reach user code. Filtering out unsupported expressions encountered before reaching the user code and erroring on unsupported expressions reached after that seemed like over-engineering a temporary work around which we were already planning to revert.

Reply:
This skipping of `__cprover_initialise` is now gone. Therefore I am going to mark this comment chain as resolved.